File: strings/7. Reverse Integer/reverse.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    is_negative = x < 0;
    num = [...Math.abs(x).toString()];
    
    for (let i = 0; i < num.length / 2; i++) {
        temp = num[i];
        num[i] = num[num.length - 1 - i];
        num[num.length - 1 - i] = temp;
    }
    
    result = parseInt(num.join(""));
    
    if (result > Math.pow(2, 31) - 1 || result < -Math.pow(2, 31)) {
        return 0;
    }
    
    return is_negative ? -result : result;
};

File: strings/7. Reverse Integer/reverse.py
class Solution:
    def reverse(self, x: int) -> int:
        is_negative = x < 0
        num = list(str(abs(x)))
        
        for i in range(len(num) // 2):
            temp = num[i]
            num[i] = num[len(num) - 1 - i]
            num[len(num) - 1 - i] = temp
        
        result = int("".join(num))
        
        if result > 2**31 - 1 or result < -2**31:
            return 0
            
        return -result if is_negative else result